experimental/GModule/Cohomology.jl

```diff
@@ -1713,9 +1713,15 @@ function Oscar.matrix(M::FreeModuleHom{FreeMod{QQAbElem}, FreeMod{QQAbElem}})
 end
 
 function ==(a::Union{Generic.ModuleHomomorphism, Generic.ModuleIsomorphism}, b::Union{Generic.ModuleHomomorphism, Generic.ModuleIsomorphism})
+  domain(a) === domain(b) || return false
+  codomain(a) === codomain(b) || return false
   return matrix(a) == matrix(b)
 end
 
+function Base.hash(a::Union{Generic.ModuleHomomorphism, Generic.ModuleIsomorphism}, h::UInt)
+  return hash(matrix(a), h)
```

Comment:
```suggestion
  h = hash(domain(a), h)
  h = hash(codomain(a), h)
  h = hash(matrix(a), h)
  return h
```
i think it would make sense to add domain/codomain to the hash as well (as it is the case in `==`)

Reply:
Also I would make sure that a `sparse_matrix` method also exists and use that. I know that these modules are dense, but their homomorphisms are probably not.

Reply:
Actually I had the code like this initially. But actually `matrix` is not defined for arbitrary `ModuleHomomorphisms`; AFAICT this code is only really used for `FPModuleHomomorphism` and then the matrix uniquely determines `domain` and `codomain`. In practice it is also harmless to have the "weaker" (but faster!) hash which does not take the domain and codomain into account: if we really have two morphisms with equal matrix but differing domain/codomain, *and* we use them, say, as keys in the same dictionary, then so be it.

So right now I'd rather lean in the opposite direction and remove these from the other `hash` method in this PR .... :-).

But in the end I really don't care too much about this either way... more important for me is that we fix the correctness bug, not how *shrug*